users/views.py

```diff
@@ -124,4 +128,26 @@ def servertime(self, *args, **kwargs):
         """ calendar endpoint.
             globally used datetime.
         """
-        return Response(self.get_server_time(), status=200)
\ No newline at end of file
+        return Response(self.get_server_time(), status=200)
+
+
+class TimeClock(Query, Slack, ViewSet):
+    """ for Clock-in and Clock-out 
+    """
+    
+    def post(self, *args, **kwargs):
+        # used by slack api
+        serializer = TimeLogSerializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        # returns if user clocked-in or clocked-out
+        instance = serializer.save()
+
+        # check instance content
+        if instance:
+            # clock in message to send back to slack
+            message = "Clocked In: " + str(datetime.datetime.now().strftime('%b %d, %Y-%H:%M:%S'))
```

Comment:
always use string interpolation instead of concatenation. forget about concatenation as it always the causes bugs and issues